### PR TITLE
Database

### PR DESCRIPTION
**Pull request by IsabellaLR**
Show stats for countries, live radar (rippling effect), highlighting countries, etc.

### CODE REVIEW

File: app/src/main/java/com/example/water/services/MapService.java [hunk 1]
```diff
@@ -47,13 +54,86 @@ public void onSuccess(Void aVoid) {
...
+    Query query = idsRef.orderBy("idNumber", Query.Direction.ASCENDING);
+
+    public static void getCountryData(String country) {
+
+        final CollectionReference countryRef = FirebaseFirestore.getInstance().collection("map").document(country).collection("cities");
+        countryRef.get().addOnCompleteListener(new OnCompleteListener<QuerySnapshot>() {
+            @Override
+            public void onComplete(@NonNull Task<QuerySnapshot> task) {
+                if (task.isSuccessful()) {
+                    Query queryAll = countryRef.orderBy("val", Query.Direction.ASCENDING);
```
Comment (jmuel) [inline app/src/main/java/com/example/water/services/MapService.java]: What is this doing? I don't see the queries referenced anywhere.

File: app/src/main/java/com/example/water/PopUpActivity.java [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+import com.jjoe64.graphview.GraphView;
+
+import java.util.ArrayList;
+
+public class PopUpActivity extends AppCompatActivity {
+
+    ViewGroup tContainer;
+    TextView city4;
```
[the remark this answers is not in the document]
Comment (IsabellaLR) [reply app/src/main/java/com/example/water/PopUpActivity.java]: shows 10 cities with worst values and I need to query from firestore first

File: app/src/main/java/com/example/water/fragments/MapFragment.java [hunk 3]
```diff
@@ -56,14 +80,16 @@
...
 public class MapFragment extends Fragment implements GoogleMap.OnCameraMoveStartedListener,
         GoogleMap.OnCameraMoveListener,
-        GoogleMap.OnCameraMoveCanceledListener,
+        GoogleMap.OnCameraMoveCanceledListener, GoogleMap.OnMapLongClickListener,
         GoogleMap.OnCameraIdleListener, OnMapReadyCallback,
         GoogleMap.OnMapClickListener {
 
-    private GoogleMap map;
-    private UiSettings uiSettings;
+    private GoogleMap mMap;
```
[the remark this answers is not in the document]
Comment (IsabellaLR) [reply app/src/main/java/com/example/water/fragments/MapFragment.java]: ok

File: app/src/main/java/com/example/water/PopUpActivity.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jmuel) [reply app/src/main/java/com/example/water/PopUpActivity.java]: The question I have is why they're hardcoded. If you need to fit them into slots in the UI then it makes sense, I'm just curious about the reasoning.

File: app/src/main/java/com/example/water/services/MapService.java (hunk 1 above)
Comment (jmuel) [reply app/src/main/java/com/example/water/services/MapService.java]: Can you describe what this is doing?

**Pull request merged**